packages/core/src/anchor/ethereum/ethereum-anchor-service.ts

```diff
@@ -87,26 +87,45 @@ export class EthereumAnchorService implements AnchorService {
   }
 
   /**
-   * Requests anchoring service for current tip of the stream
+   * Send request to the anchoring service
+   * @param carFile - CAR file containing all necessary data for the CAS to anchor
+   * @param waitForConfirmation - if true, waits until the CAS has acknowledged receipt of the anchor
+   *   request before returning.
    */
-  requestAnchor(carFile: CAR): Observable<CASResponse> {
+  async requestAnchor(
+    carFile: CAR,
+    waitForConfirmation: boolean
+  ): Promise<Observable<CASResponse>> {
     const carFileReader = new AnchorRequestCarFileReader(carFile)
     const cidStreamPair: CidAndStream = { cid: carFileReader.tip, streamId: carFileReader.streamId }
-    return concat(
+
+    const requestCreated$ = concat(
       this._announcePending(cidStreamPair),
-      this._makeAnchorRequest(carFileReader),
-      this.pollForAnchorResponse(carFileReader.streamId, carFileReader.tip)
-    ).pipe(
-      catchError((error) =>
-        of<CASResponse>({
-          id: '',
-          status: AnchorRequestStatusName.FAILED,
-          streamId: carFileReader.streamId,
-          cid: carFileReader.tip,
-          message: error.message,
-        })
-      )
+      this._makeAnchorRequest(carFileReader, !waitForConfirmation)
     )
+
+    const anchorCompleted$ = this.pollForAnchorResponse(carFileReader.streamId, carFileReader.tip)
+
+    const addErrorHandling = (anchorStatus$) => {
+      return anchorStatus$.pipe(
+        catchError((error) =>
+          of<CASResponse>({
+            id: '',
+            status: AnchorRequestStatusName.FAILED,
+            streamId: carFileReader.streamId,
+            cid: carFileReader.tip,
+            message: error.message,
+          })
+        )
+      )
+    }
+
+    if (waitForConfirmation) {
+      await lastValueFrom(requestCreated$)
+      return addErrorHandling(anchorCompleted$)
```

Comment:
Nitpick suggestion:

```
const streamId = carFileReader.streamId
const tip = carFileReader.tip
const handleErrors = catchError((error) => 
  of<CASResponse>({
            id: '',
            status: AnchorRequestStatusName.FAILED,
            streamId: streamId,
            cid: tip,
            message: error.message,
          })
)

return anchorCompleted$.pipe(handleErrors)
```

and few lines below:
```
      return concat(requestCreated$, anchorCompleted$).pipe(handleErrors)
```

Reply:
I got a compiler error trying to bind catchError to a variable, but I did something similar with the variable just holding the inner error handler and then adding the pipe to catcherror in both places